File: 17-missing-rust/samples/Pi.java
public final class Pi {
  public static void main(String[] args) {
    int n = 355, d = 113, whole = n / d, rem = n % d;
    StringBuilder digits = new StringBuilder();
    for (int i = 0; i < 4; i++) {
      rem *= 10;
      digits.append(rem / d);
      rem %= d;
    }
    System.out.println("hello " + whole + "." + digits + " world!");
  }
}

File: 17-missing-rust/samples/Pi.kt
fun main() {
    val n = 355
    val d = 113
    val whole = n / d
    var rem = n % d
    val digits = StringBuilder()
    repeat(4) {
        rem *= 10
        digits.append(rem / d)
        rem %= d
    }
    println("hello $whole.$digits world!")
}
